src/MSBuild/XMake.cs

```diff
@@ -89,6 +89,13 @@ public enum ExitType
             ProjectCacheFailure
         }
 
+#if FEATURE_OSVERSION
+        /// <summary>
+        /// True if the Main method was invoked. False indicates that we're running hosted in another process (e.g. unit tests).
+        /// </summary>
+        private static bool s_executingMainEntryPoint;
```

Comment:
It looks like this variable is set once and used once—in a function that executes once.

Instead of a new variable, what if we inlined it? In Main:
```C#
InitializationException.VerifyThrow(Environement.OSVersion.Platform == PlatformID.Win32NT && Environment.OSVersion.Version.Major >= 10, "UnsupportedOS");
```

and in Execute:
```C#
InitializationException.VerifyThrow(Environement.OSVersion.Platform == PlatformID.Win32NT && (Environment.OSVersion.Version.Major > 6 || (Environment.OSVersion.Version.Major == 6 && Environment.OSVersion.Version.Minor >= 2), "UnsupportedOS");
```

Reply:
I didn't want to move the check to not break something. What you're suggesting would be fine I think, thank you!

I have removed the variable and made us use Win7 as the minimum version, per the rationale [above](https://github.com/dotnet/msbuild/pull/6877#issuecomment-929973564).